Source/Details/_ASCollectionReusableView.h

```diff
@@ -18,15 +18,41 @@
 #import <UIKit/UIKit.h>
 #import <AsyncDisplayKit/ASBaseDefines.h>
 
-@class ASCellNode, ASCollectionElement;
+@class ASCollectionElement;
+
+/**
+ * Attempts to cast x to _ASCollectionReusableView, or nil if not possible.
+ *
+ * Since _ASCollectionReusableView is not available for subclassing (see below),
+ * comparing x's and _ASCollectionReusableView's classes is faster than calling -isKindOfClass: on x.
+ */
+#define ASCollectionReusableViewCast(x) ({ \
+  id __var = x; \
+  ((_ASCollectionReusableView *) (x.class == [_ASCollectionReusableView class] ? __var : nil)); \
+})
```

Comment:
Regarding the class hierarchy, note that the UICV case has subclasses in the app too. Something like (guessing at the hierarchy):

CustomCellClass -> UICollectionViewCell -> UICollectionReusableView -> UIView -> UIResponder -> NSObject